OWML.ModLoader/ModData.cs

```diff
@@ -17,15 +15,5 @@ public ModData(IModManifest manifest, IModConfig config, IModConfig defaultConfi
             DefaultConfig = defaultConfig;
         }
 
-        public void ResetConfig()
```

Comment:
Removing this is a breaking change but I think no mods used it. Did any of you?

Reply:
I actually don't *have* to remove this any more, after introducing the base class... kinda wanna do it still 🤷

Reply:
I suppose resetting to defaults is handled by `ModConfig` now?

Reply:
Yeah, which do you think should do it - ModData or ModConfigMenu?

Reply:
I think it depends whether or not you want to implement "Reset all mods to defaults" feature

Reply:
Don't really see the point of that feature... do you?

Reply:
not really...
BTW, when an item gets removed/added to default config do you automatically push that change to editable config?

Reply:
Only reason I see is if something goes wrong after some updated mod gets confused by outdated config and user can't locate which one it is

Reply:
If a value is added to the "settings" list/dict in default config, that isn't in config, it'll be added to config. Changes/removals to/of existing values won't be updated to config, until the user resets to default.

Reply:
If you think there's value in keeping the reset method in mod data, I'll revert it. I don't have a strong opinion on this. Give the word :)

Reply:
Only value I can see is making implementing "Reset All to Defaults" easier. And only reason I can see why user might want to reset all is if there are problems with outdated config and there is either multiple of them or user can't find which one is misbehaving

Reply:
BTW, I suggest you to think about picking up type changes from default config (i.e. number to slider or to enum)

Reply:
And warning on (seemingly) incompatible types

Reply:
That sounds hard. Personally I don't think it would be worth it. But either way I'd argue it's outside the scope of this PR :)